### DISCUSSION TITLE
Graphcache - DetailView -> ListView

### DISCUSSION THREAD

**Discussion by szv** [urql-graphql/urql#2365] [Q&A] [answered]
Hello,

first of all thanks for creating URQL 👏 

I am using:
- Angular 13
- @urql/core 2.4.3
- @urql/exchange-graphcache 4.3.6

I configured the client like this:
```TypeScript
createClient({
    url: environment.graphQLUri,
    exchanges: [
        devtoolsExchange,
        dedupExchange,
        cacheExchange({                                // https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/
            schema: schema,
            keys: {
                CollectionSegmentInfo: data => null,
                CompanyCollectionSegment: data => null,
                UserCollectionSegment: data => null
            },
            resolvers: {
                Query: {
                    companyById: (_, args) => ({ __typename: 'Company', id: args.id }),
                    userById: (_, args) => ({ __typename: 'User', id: args.id }),
                },
            }
        }),
        fetchExchange
    ]
});
```

Currently I am facing the following problem:  
I have an application with a _listview_ and a _detailview_ of companies. When loading the application in the _listview_ it loads the companies and when navigating to the _detailview_ it loads the details of the specified company. So far so good.  
(Btw I am using @defer to get partial results)

But when loading the application in the _detailview_ it loads the specified company and when navigation to the _listview_ it does not show any results. No network request is make eigther.   
When navigating to the users listview it does not load anything eigther.

In the URQL-DevTools-Explorer it says:
- companyById(id: 1)  cache miss (of course)
- companiesOffsetPaged: undefined
- usersOffsetPaged: undefined

Do you have any ideas on this?

Kind regards

EDIT:  
I just noticed, that this only occurs, when the ```requestPolicy``` in the client-configuration is set to ```cache-first``` aka not set (as in the code-snippet above).  
When setting the ```requestPolicy``` in the client-configuration to ```cache-and-network``` to problem described above does not occur.  
Setting the ```requestPolicy``` to ```cachen-and-network``` only for a specific operation did not help.

**Comment by haswalt** [comment]
I'm assuming that when you navigate back to the list view the browser is using a cached page. You'll want to use something like the refocus exchange to re-run queries when a view is shown: https://formidable.com/open-source/urql/docs/api/refocus-exchange/.

Alternatively you could use whatever mechanism your router provides to detect the route changing to the list view and re-run the query then.

**Reply by szv** [reply]
@haswalt Thanks for your reply :)

I think you misunterstood my problem.  

1. The default requestPolicy is ```cache-first```
2. The user loads the Angular-application in the detailview e.g. ```/comapnies/1``` - _so this is the first visit of the website and no query has been executed yet, which means the cache is empty_
3. The query for the detailview gets executed - _up to here, all good_
4. Now the user navigates to a listview e.g. ```/companies``` or ```/users```
5. From now on, no query gets performed
6. The URQL-DevTool says, that the result of the queries for the listviews are ```undefined``` - _as shown in the screenshot_
![urql-devtools](https://user-images.githubusercontent.com/10291845/160644444-7b2806ea-ded0-49cf-95df-971a4aff4920.png)


Do you have any ideas on that? 🥺

**Comment by szv** [comment, marked as answer]
It was fixed with #2386  aka  @urql/core v2.4.4
